submissions/the-darkesttt/frogger-task/app.js

```diff
@@ -0,0 +1,323 @@
+    
+            this.menuList.appendChild(this.menuItem);
+            this.menuItem.appendChild(itemImg);
+            this.menuItem.appendChild(imgDesc);
+            
+        });
+
+        document.body.appendChild(this.bg);
+        document.body.appendChild(this.menu);
+
+        this.menu.appendChild(this.menuTitle);
+        this.menu.appendChild(this.menuDescriprion);
+        this.menu.appendChild(this.menuList);
+
+        this.menuList.addEventListener('click', (event) => {
+            const listTarget = event.target;
+            if (listTarget.tagName == "UL") return;
+        
+            player.sprite = listTarget.getAttribute("src");
+    
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+        });
+    }
+
+    death(points) {
+        if (this.menuList.parentElement != null) {
+            this.menuList.parentElement.removeChild(this.menuList);
+        }
+
+        this.bg.classList.remove('menu-animation');
+        this.menu.classList.remove('menu-animation');
+
+        this.menuDescriprion.innerText = 'You died! Your score:';
+
+        this.menuScore = document.createElement('span');
+        this.menuScore.innerText = points;
+        this.menuScore.classList.add('menu-score');
+
+        this.menu.appendChild(this.menuScore);
+
+        this.menuClose = document.createElement('button');
+        this.menuClose.innerText = 'Close menu and continue';
+        this.menuClose.classList.add('menu-close');
+        this.menu.appendChild(this.menuClose);
+
+        this.menuClose.addEventListener('click', (event) => {
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+            this.menuDescriprion.innerText = '';
+            this.menuScore.parentElement.removeChild(this.menuScore);
+            this.menuClose.parentElement.removeChild(this.menuClose);
+        });
+    }
+}
+
+
+const startMenu = new Menu();
+startMenu.start();
+
+let points = 0;
+const score = document.createElement('p');
+score.classList.add('score');
+document.body.appendChild(score);
+
+function updateScore() {
+    score.innerHTML = 'Score: ' + points;
+}
+
+updateScore();
+
+function animateScoreGreen() {
+    score.classList.add('green');
+    setTimeout(function() {
+        score.classList.remove('green');
+    }, 1000)
+}
+
+function animateScoreBlue() {
+    score.classList.add('blue');
+    setTimeout(function() {
+        score.classList.remove('blue');
+    }, 1000)
+}
+
+let health = 3;
+
+const healthBar = document.createElement('div');
+document.body.appendChild(healthBar);
+healthBar.classList.add("health");
+
+function generateHealthBar() {
+
+    if (health <= 0) {
+        death();
+    }
+
+    if (healthBar.children.length > 0){
+        let imgArr = Array(...healthBar.children);
+        imgArr.forEach(item => {
+            item.remove();
+        });
+    }
+
+    for (let i = 1; i <= health; i++) {
```

Comment:
Better go 
```js
for (let i = 0; i < health; i++) {...
```
When you use for and starting `i` doesn't equal `0` it draws attention to why is that. I believe because you had 4 hearts instead of 3 you are starting with `1` instead of `0`. Your problem was that `<=`. Just use `<` and `i` will be `0, 1, 2` and not `0, 1, 2, 3`.